File: Tests/LetterChange/LetterChange.cs
namespace Tests.MultiplyBy
{
    using System.Collections.Generic;
    using System.Linq;
    using CSharp.Codewars;
    using FSharp.CodeWars;
    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class LetterChangeFsTests
    {
        [TestMethod]
        public void LetterChangeFs_JavaScript()
        {
            Assert.AreEqual("KbwbTdsjqu", LetterChangeFs.letterChange("JavaScript"));
        }

        [TestMethod]
        public void LetterChangeFs_LoremIpsum()
        {
            Assert.AreEqual("Mpsfn Jqtvn", LetterChangeFs.letterChange("Lorem Ipsum"));
        }

        [TestMethod]
        public void LetterChangeFs_CornerZ_A()
        {
            Assert.AreEqual("A", LetterChangeFs.letterChange("Z"));
        }

        [TestMethod]
        public void LetterChangeFs_Cornerz_a()
        {
            Assert.AreEqual("A", LetterChangeFs.letterChange("Z"));
        }
    }

    [TestClass]
    public class LetterChangeCsTests
    {
        [TestMethod]
        public void LetterChangeCs_JavaScript()
        {
            Assert.AreEqual("KbwbTdsjqu", LetterChangeCs.LetterChange("JavaScript"));
        }

        [TestMethod]
        public void LetterChangeCs_LoremIpsum()
        {
            Assert.AreEqual("Mpsfn Jqtvn", LetterChangeCs.LetterChange("Lorem Ipsum"));
        }

        [TestMethod]
        public void LetterChangeCs_CornerZ_A()
        {
            Assert.AreEqual("A", LetterChangeCs.LetterChange("Z"));
        }

        [TestMethod]
        public void LetterChangeCs_Cornerz_a()
        {
            Assert.AreEqual("A", LetterChangeCs.LetterChange("Z"));
        }
    }
}

File: Javascript/beta/letterChange.js
function letterChange(str) {

    function nextChar(c) {
        switch (c) {
            case 'Z': return 'A';
            case 'z': return 'a';
            case ' ': return ' ';
            default: return String.fromCharCode(c.charCodeAt(0) + 1);
        };
    };

    return str.split("").map(nextChar).join("");
};

describe("letterChange - ", function () {
    it("JavaScript", function () { expect(letterChange("JavaScript")).toEqual("KbwbTdsjqu"); });
    it("Lorem Ipsum", function () { expect(letterChange("Lorem Ipsum")).toEqual("Mpsfn Jqtvn"); });
    it("Z", function () { expect(letterChange("Z")).toEqual("A"); });
    it("z", function () { expect(letterChange("z")).toEqual("a"); });
});